### PR TITLE
[DEV-951] add balance route

### PR DESCRIPTION
**Pull request by Yannick1712**

### CODE REVIEW

File: src/subdomains/staking/api/controllers/staking.controller.ts [hunk 1]
```diff
@@ -23,6 +24,13 @@ export class StakingController {
     return this.stakingService.getOrCreateStaking(jwt.userId, jwt.walletId, query);
   }
 
+  @Get('deposit-address')
```
Comment (davidleomay) [inline src/subdomains/staking/api/controllers/staking.controller.ts:27]: `@Get('balance')`? (we want to get the balance)

File: src/subdomains/staking/application/services/staking.service.ts [hunk 2]
```diff
@@ -65,6 +66,16 @@ export class StakingService {
     return StakingOutputDtoMapper.entityToDto(await this.authorize.authorize(userId, existingStaking.id));
   }
 
+  async getDepositAddressBalance(address: string): Promise<DepositAddressBalanceOutputDto> {
+    const stakingEntity = await this.repository.findOne({
+      where: { depositAddress: { address: address } },
+      relations: ['depositAddress'],
+    });
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:73]: What do you get here, if address is undefined?

File: src/subdomains/staking/application/services/staking.service.ts [hunk 3]
```diff
@@ -65,6 +66,16 @@ export class StakingService {
...
+    if (!stakingEntity?.balance && stakingEntity?.balance !== 0)
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:74]: Weird check? What about this?
```ts
if (!stakingEntity) throw ...
```

File: src/subdomains/staking/application/services/staking.service.ts (hunk 2 above)
Comment (Yannick1712) [reply src/subdomains/staking/application/services/staking.service.ts:73]: undefined 👍

**Review by davidleomay — APPROVED** [review]

**Pull request merged**